Added Explaination on Data Model (#8)
icure/medtech-docs: sdks/explanations/data-model/data-model-healthcare-element.md

---
slug: healthcare-element
tags:
    - data model
    - healthcare element
---
# Healthcare Element

A Healthcare Element is a piece of medical information that can give some context about the status of a Patient or to a
series of measurements. 

## When to Use a Healthcare Element?

You should use a Healthcare Element when a [Data Owner](/sdks/glossary#data-owner) wants to specify an underlying 
condition related to a Patient, to a visit or to a set of measurements associated to a Patient.

## How a Healthcare Element is Related to Other Entities?

The medical information contained in the Healthcare Element can be specified using a standard terminology, such as 
SNOMED CT, by means of a Coding.  
A Healthcare Element can be associated to one or more Data Samples to give a context to their measurements.  
A Healthcare Element can use a Coding to associate its medical information to the codes of a [Terminology](http://localhost:3000/sdks/glossary#terminologies).

## Examples

### A Patient Registering Their Symptoms

> A Patient has a period tracking app and wants to add some symptoms she is experiencing (headache) by adding a new Data Sample.
> Then, she adds the information that her period started as a Healthcare Element associated to the Data Sample.

Reviewer: The example shows stuff the other way around: first create the healthcare element then create the data sample:
```suggestion
A Patient has a period tracking app and wants to register that her period started and that she is experiencing a headache.
She will create a Healthcare Element to register that her period started, then she will create a Data Sample associated to the Healthcare Element to register the headache.
```

<!-- file://code-samples/explanation/patient-creates-data-sample/index.mts snippet:patient can create DS and HE-->
```typescript
const healthcareElement = await api.healthcareElementApi.createOrModifyHealthcareElement(
    new HealthcareElement({
        description: 'My period started'
    }),
    patient.id
)

await api.dataSampleApi.createOrModifyDataSampleFor(
    patient.id,
    new DataSample({
        content: { 'en': new Content({
                stringValue: 'I have a headache'
            })},
        healthcareElementIds: new Set([healthcareElement.id])
    })
)
```

### A Doctor Updating the Status of a Patient

> A Doctor (Healthcare Professional) discovers that their Patient is pregnant. Therefore, it updates her condition.

Reviewer: ```suggestion
A Doctor (Healthcare Professional) discovers that their Patient is pregnant. Therefore, they update her condition.
```
